hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/protocolPB/OzoneManagerProtocolClientSideTranslatorPB.java

```diff
@@ -1004,6 +1006,46 @@ public List<OmKeyInfo> listKeys(String volumeName, String bucketName,
 
   }
 
+  /**
+   * Lightweight listKeys implementation.
+   */
+  @Override
+  public List<BasicOmKeyInfo> listKeysLight(String volumeName,
+      String bucketName, String startKey, String prefix,
+      int maxKeys) throws IOException {
+    List<BasicOmKeyInfo> keys = new ArrayList<>();
+    ListKeysRequest.Builder reqBuilder = ListKeysRequest.newBuilder();
+    reqBuilder.setVolumeName(volumeName);
+    reqBuilder.setBucketName(bucketName);
+    reqBuilder.setCount(maxKeys);
+
+    if (startKey != null) {
+      reqBuilder.setStartKey(startKey);
+    }
+
+    if (prefix != null) {
+      reqBuilder.setPrefix(prefix);
+    }
+
+    ListKeysRequest req = reqBuilder.build();
+
+    OMRequest omRequest = createOMRequest(Type.ListKeysLight)
+            .setListKeysRequest(req)
+            .build();
+
+    ListKeysLightResponse resp =
+            handleError(submitRequest(omRequest)).getListKeysLightResponse();
+    List<BasicOmKeyInfo> list = new ArrayList<>();
```

Comment:
Why did you create another `list` object and not directly used `keys`?

Reply:
I have removed the duplicate `list` object, thanks!